### PR TITLE
Measure Model

### PR DESCRIPTION
**Pull request by hossenlopp**
Pull requests into cqm-models require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/BONNIE-1575
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] If there were any JavaScript changes, this PR has updated the `dist` directory using `npm run dist`
- [x] If applicable, the library version number in `package.json` and `cqm-models.gemspec` has been updated
- [x] All changes can be reproduced by running the generator script N/A
- [ ] Cqm-execution fixtures have been updated with the update_cqm_execution_fixtures.sh script inside server-scripts using this branch in the cqm-converter

**Bonnie Reviewer:**

Name: @daco101 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code
- [x] You have executed the generator script, and made sure no changes were made that the generator did not reproduce itself

**Cypress Reviewer:**

Name: @mayerm94 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code
- [x] You have executed the generator script, and made sure no changes were made that the generator did not reproduce itself

### CODE REVIEW

File: app/assets/javascripts/cqm/CQLStatementDependency.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  {
+    library_name: String,
+    statement_name: String,
+  }
+);
+
+const StatementDependencySchema = new mongoose.Schema(
+  {
+    statement_name: String,
+    dependencies: [StatementReferenceSchema],
```
Comment (daco101) [inline app/assets/javascripts/cqm/CQLStatementDependency.js]: Possibly this should still be a hash instead of an array, its looks like the frontend looks up dependencies per statement.

File: app/assets/javascripts/cqm/PopulationSet.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    id: String,
+    statement: StatementReferenceSchema
+  }
+);
+
+const ObservationSchema = new mongoose.Schema(
+  {
+    title: String,
+    observation_function: StatementReferenceSchema,
+    aggregation_typee: String
```
Comment (mayerm94) [inline app/assets/javascripts/cqm/PopulationSet.js]:
```suggestion
    aggregation_type: String
```

File: app/assets/javascripts/cqm/ValueSet.js [hunk 3]
```diff
@@ -7,18 +7,15 @@ const [String, Mixed, ObjectId] = [
...
     display_name: String,
     version: String,
-    categories: Mixed,
 
     concepts: [Concept.ConceptSchema],
 
-    user: { type: ObjectId, ref: 'User', index: true }, // Bonnie-specific
-    bundle: { type: ObjectId, ref: 'Bundle' }, // Cypress-specific
-    measures: [{ type: ObjectId, ref: 'Measure' }],
+    measures: [{ type: ObjectId, ref: 'Measure', index: true }],
```
Comment (daco101) [inline app/assets/javascripts/cqm/ValueSet.js]: Is this needed given the `has_and_belongs_to_many :value_sets, inverse_of: nil` in the measure model?

Comment (hossenlopp) [reply app/assets/javascripts/cqm/ValueSet.js]: Removed it.
